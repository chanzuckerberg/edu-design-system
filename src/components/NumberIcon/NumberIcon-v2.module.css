@import '../../design-tokens/mixins.css';
/*------------------------------------*\
    # NUMBER ICON
\*------------------------------------*/

/**
 * Number Icon displays a number enclosed in a circle.
 * 
 * Centers the number text in the circle.
 */
.number-icon {
  /* Line height set to 1 here since this should only ever be on 1 line and it evens out padding in circle. */
  line-height: 1;
  display: flex;
  justify-content: center;
  align-items: center;

  /* The circle part of the icon, made with borders. */
  border: var(--eds-border-width-sm) solid;
  border-color: inherit;
  border-radius: var(--eds-border-radius-full);

  cursor: pointer;
}

/**
 * Size Variants.
 */
.number-icon--size-md {
  height: 1.5rem;
  width: 1.5rem;
  min-width: 1.5rem;
}

.number-icon--size-lg {
  height: 2rem;
  width: 2rem;
  min-width: 2rem;
}

/* Colors & Theme */

/**
 * Interactive States
 */
.number-icon--status-default {
  color: var(--eds-theme-color-text-utility-interactive-primary);
  border-color: var(--eds-theme-color-border-utility-interactive);
  background-color: var(--eds-theme-color-background-utility-interactive-no-emphasis);

  &:hover {
    border-color: var(--eds-theme-color-border-utility-interactive-hover);
    background-color: var(--eds-theme-color-background-utility-interactive-no-emphasis-hover);
  }

  &:active {
    border-color: var(--eds-theme-color-border-utility-interactive-active);
    background-color: var(--eds-theme-color-background-utility-interactive-no-emphasis-active);
  }
}

.number-icon--status-completed {
  color: var(--eds-theme-color-text-utility-inverse);
  border-color: var(--eds-theme-color-background-utility-favorable-high-emphasis);
  background-color: var(--eds-theme-color-background-utility-favorable-high-emphasis);

  &:hover {
    border-color: var(--eds-theme-color-background-utility-favorable-high-emphasis-hover);
    background-color: var(--eds-theme-color-background-utility-favorable-high-emphasis-hover);
  }

  &:active {
    border-color: var(--eds-theme-color-background-utility-favorable-high-emphasis-active);
    background-color: var(--eds-theme-color-background-utility-favorable-high-emphasis-active);
  }
}

.number-icon--status-incomplete {
  color: var(--eds-theme-color-text-utility-neutral-secondary);
  border-color: var(--eds-theme-color-border-utility-neutral-medium-emphasis);

  border-style: dashed;
  pointer-events: none;
}