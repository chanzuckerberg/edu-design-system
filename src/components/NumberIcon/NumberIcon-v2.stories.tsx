import type { StoryObj, Meta } from '@storybook/react';
import React from 'react';

import { NumberIcon } from './NumberIcon-v2';

export default {
  title: 'Components/V2/NumberIcon',
  component: NumberIcon,
  parameters: {
    badges: ['intro-1.0', 'current-2.0'],
  },
  args: {
    'aria-label': 'number icon example',
    number: 1,
  },
  decorators: [(Story) => <div className="p-8">{Story()}</div>],
} as Meta<Args>;

type Args = React.ComponentProps<typeof NumberIcon>;
type Story = StoryObj<Args>;

export const Default: Story = {};

export const Sizes: Story = {
  args: {
    status: 'default',
  },
  render: (args) => {
    return (
      <>
        <NumberIcon number={2} size="md" {...args} />
        <NumberIcon number={3} size="lg" {...args} />
      </>
    );
  },
  decorators: [
    (Story) => <div className="flex flex-wrap gap-1">{Story()}</div>,
  ],
};

export const Completed: Story = {
  args: {
    ...Sizes.args,
    status: 'completed',
  },
  render: Sizes.render,
  decorators: Sizes.decorators,
};

export const Incomplete: Story = {
  args: {
    ...Sizes.args,
    status: 'incomplete',
  },
  render: Sizes.render,
  decorators: Sizes.decorators,
};

/**
 * `NumberIcon` supports individual digits, with a maximum of two digits. By default,
 * they are positioned as block-level elements. use `flex` or `display` to update positioning.
 */
export const DifferentNumbers: Story = {
  /**
   * Disables controls for args that have no affect on this story
   */
  argTypes: {
    number: {
      table: {
        disable: true,
      },
    },
    'aria-label': {
      table: {
        disable: true,
      },
    },
  },
  render: (args) => (
    <div>
      {[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 21, 32, 43, 54, 65, 76, 87, 98].map(
        (number) => (
          <NumberIcon
            key={number}
            {...args}
            aria-label={`Step ${number}`}
            number={number}
          />
        ),
      )}
    </div>
  ),
};

/**
 * This Implementation example shows how to use Number Icon to build a stepper-like component.
 *
 * - incomplete rows are aligned with each number icon to show progress
 */
export const NumberIconList: Story = {
  parameters: {
    badges: ['intro-1.0', 'current-2.0', 'implementationExample'],
  },
  render: () => (
    <div className="flex flex-wrap gap-1">
      <NumberIcon aria-label="Item 1" number={1} size="md" />
      <NumberIcon aria-label="Item 2" number={2} size="md" />
      <NumberIcon aria-label="Item 3" number={3} size="md" />
      <NumberIcon aria-label="Item 4" number={4} size="md" />
      <NumberIcon aria-label="Item 5" number={5} size="md" />
      <NumberIcon aria-label="Item 6" number={6} size="md" />
    </div>
  ),
};
