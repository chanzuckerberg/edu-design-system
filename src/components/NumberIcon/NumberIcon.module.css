@import '../../design-tokens/mixins.css';
/*------------------------------------*\
    # NUMBER ICON
\*------------------------------------*/

/**
 * Number Icon displays a number enclosed in a circle.
 * 
 * Centers the number text in the circle.
 */
.number-icon {
  /* Line height set to 1 here since this should only ever be on 1 line and it evens out padding in circle. */
  line-height: 1;
  display: flex;
  justify-content: center;
  align-items: center;

  /* The circle part of the icon, made with borders. */
  border: calc(var(--eds-border-width-sm) * 1px) solid;
  border-color: inherit;
  border-radius: calc(var(--eds-border-radius-full) * 1px);

  &.number-icon--is-interactive {
    cursor: pointer;
  }
}

/**
 * Size Variants.
 */
.number-icon--size-md {
  height: calc(var(--eds-size-3) / 16 * 1rem);
  width: calc(var(--eds-size-3) / 16 * 1rem);
  min-width: calc(var(--eds-size-3) / 16 * 1rem);
}

.number-icon--size-lg {
  height: calc(var(--eds-size-4) / 16 * 1rem);
  width: calc(var(--eds-size-4) / 16 * 1rem);
  min-width: calc(var(--eds-size-4) / 16 * 1rem);
}

/* Colors & Theme */

/**
 * Interactive States
 */
.number-icon--status-default {
  color: var(--eds-theme-color-text-utility-interactive-primary);
  border-color: var(--eds-theme-color-border-utility-interactive);
  background-color: var(--eds-theme-color-background-utility-interactive-no-emphasis);

  &.number-icon--is-interactive {
    &:hover {
      border-color: var(--eds-theme-color-border-utility-interactive-hover);
      background-color: var(--eds-theme-color-background-utility-interactive-no-emphasis-hover);
    }

    &:active {
      border-color: var(--eds-theme-color-border-utility-interactive-active);
      background-color: var(--eds-theme-color-background-utility-interactive-no-emphasis-active);
    }
  }
}

.number-icon--status-completed {
  color: var(--eds-theme-color-text-utility-inverse);
  border-color: var(--eds-theme-color-background-utility-favorable-high-emphasis);
  background-color: var(--eds-theme-color-background-utility-favorable-high-emphasis);

  &.number-icon--is-interactive {
    &:hover {
      border-color: var(--eds-theme-color-background-utility-favorable-high-emphasis-hover);
      background-color: var(--eds-theme-color-background-utility-favorable-high-emphasis-hover);
    }

    &:active {
      border-color: var(--eds-theme-color-background-utility-favorable-high-emphasis-active);
      background-color: var(--eds-theme-color-background-utility-favorable-high-emphasis-active);
    }
  }
}

.number-icon--status-incomplete {
  color: var(--eds-theme-color-text-utility-default-secondary);
  border-color: var(--eds-theme-color-border-utility-default-medium-emphasis);

  border-style: dashed;
  pointer-events: none;
}