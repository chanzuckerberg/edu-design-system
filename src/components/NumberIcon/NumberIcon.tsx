import clsx from 'clsx';
import React from 'react';
import styles from './NumberIcon.module.css';
import Icon from '../Icon';
import Text from '../Text';

export interface Props {
  /**
   * Screen-reader text for the number icon.
   */
  'aria-label': string;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  /**
   * Incomplete prop to show incomplete state
   */
  incomplete?: boolean;
  /**
   * Number to be shown as the icon.
   */
  number?: number;
  /**
   * The size of the icon. Defaults to 'lg'.
   */
  size?: 'sm';
  /**
   * The color variant of the icon. Defaults to 'base'.
   */
  variant?: 'base' | 'success';
  /**
   * Number icon title
   */
  numberIconTitle?: string;
}

/**
 * ```ts
 * import {NumberIcon} from "@chanzuckerberg/eds";
 * ```
 *
 * Encloses a number in a circle to be used as an icon.
 *
 * Example usage:
 *
 * ```tsx
 * <NumberIcon aria-label="1" number={1} />
 * ```
 */
export const NumberIcon = ({
  className,
  number,
  size,
  variant = 'base',
  incomplete,
  numberIconTitle,
  ...other
}: Props) => {
  const componentClassName = clsx(
    styles['number-icon'],
    styles[`number-icon--${variant}`],
    size && styles[`number-icon--${size}`],
    incomplete && styles['number-icon--incomplete'],
    className,
  );

  /**
   * 1) Set as 'span' since icon use is more inline than block, but no effect since display is 'flex'.
   * 2) When `incomplete` is defined and there is a numberIconTitle on the circle icon, then this will render
   * the proper icon with the incomplete text provided to that icon.
   * 3) If this is not incomplete, then the number prop provided will show within the border
   */
  return (
    <Text
      as="span" /* 1 */
      className={componentClassName}
      role="img"
      size="sm"
      variant={variant === 'base' ? 'neutral-medium' : 'white'}
      {...other}
    >
      {incomplete && numberIconTitle ? (
        <Icon
          className={styles['number-icon__icon']}
          color="inherit"
          name="circle"
          purpose="informative"
          size={size === 'sm' ? '0.25rem' : '0.5rem'}
          title={numberIconTitle}
        />
      ) : (
        number
      )}
    </Text>
  );
};
