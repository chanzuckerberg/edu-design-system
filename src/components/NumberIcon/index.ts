export { NumberIcon as default } from './NumberIcon';
export { NumberIcon as NumberIconV2 } from './NumberIcon-v2';
