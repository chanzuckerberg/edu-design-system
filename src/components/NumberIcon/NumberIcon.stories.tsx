import type { StoryObj, Meta } from '@storybook/react';
import React from 'react';

import { NumberIcon } from './NumberIcon';

export default {
  title: 'Components/NumberIcon',
  component: NumberIcon,
  parameters: {
    layout: 'centered',
  },
  args: {
    'aria-label': 'number icon example',
    number: 1,
  },
  decorators: [(Story) => <div className="p-spacing-size-4">{Story()}</div>],
  tags: ['autodocs', 'version:2.0'],
} as Meta<Args>;

type Args = React.ComponentProps<typeof NumberIcon>;
type Story = StoryObj<Args>;

export const Default: Story = {};

export const Sizes: Story = {
  args: {
    status: 'default',
  },
  render: (args) => {
    return (
      <>
        <NumberIcon number={2} size="md" {...args} />
        <NumberIcon number={3} size="lg" {...args} />
      </>
    );
  },
  decorators: [
    (Story) => <div className="flex flex-wrap gap-1">{Story()}</div>,
  ],
};

/**
 * `NumberIcon` can be used in interactive contexts, when wrapped by a navigable or interactive element.
 */
export const IsInteractive: Story = {
  args: {
    ...Sizes.args,
    isInteractive: true,
  },
  render: Sizes.render,
};

export const Completed: Story = {
  args: {
    ...Sizes.args,
    status: 'completed',
  },
  render: Sizes.render,
  decorators: Sizes.decorators,
};

export const Incomplete: Story = {
  args: {
    ...Sizes.args,
    status: 'incomplete',
  },
  render: Sizes.render,
  decorators: Sizes.decorators,
};

/**
 * `NumberIcon` supports individual digits, with a maximum of two digits. By default,
 * they are positioned as block-level elements. use `flex` or `display` to update positioning.
 */
export const DifferentNumbers: Story = {
  /**
   * Disables controls for args that have no affect on this story
   */
  argTypes: {
    number: {
      table: {
        disable: true,
      },
    },
    'aria-label': {
      table: {
        disable: true,
      },
    },
  },
  render: (args) => (
    <div>
      {[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 21, 32, 43, 54, 65, 76, 87, 98].map(
        (number) => (
          <NumberIcon
            key={number}
            {...args}
            aria-label={`Step ${number}`}
            number={number}
          />
        ),
      )}
    </div>
  ),
};

/**
 * This Implementation example shows how to use Number Icon to build a stepper-like component.
 *
 * - incomplete rows are aligned with each number icon to show progress
 */
export const NumberIconList: Story = {
  tags: ['code-only'],
  render: () => (
    <div className="flex flex-wrap gap-1">
      <NumberIcon aria-label="Item 1" number={1} size="md" />
      <NumberIcon aria-label="Item 2" number={2} size="md" />
      <NumberIcon aria-label="Item 3" number={3} size="md" />
      <NumberIcon aria-label="Item 4" number={4} size="md" />
      <NumberIcon aria-label="Item 5" number={5} size="md" />
      <NumberIcon aria-label="Item 6" number={6} size="md" />
    </div>
  ),
};
