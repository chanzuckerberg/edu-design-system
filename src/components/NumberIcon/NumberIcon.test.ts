import { generateSnapshots } from '@chanzuckerberg/story-utils';
import type { StoryFile } from '@storybook/testing-react';
import * as stories from './NumberIcon.stories';

describe('<NumberIcon />', () => {
  generateSnapshots(stories as StoryFile);
});
