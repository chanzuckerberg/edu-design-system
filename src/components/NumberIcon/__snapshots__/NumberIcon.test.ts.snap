// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberIcon /> Default story renders snapshot 1`] = `
<div
  class="p-8"
>
  <span
    aria-label="Step 1"
    class="text text--body-sm number-icon number-icon--base number-icon--lg"
    role="img"
  >
    1
  </span>
</div>
`;

exports[`<NumberIcon /> DifferentNumbers story renders snapshot 1`] = `
<div
  class="p-8"
>
  <div>
    <span
      aria-label="Step 0"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      0
    </span>
    <span
      aria-label="Step 1"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      1
    </span>
    <span
      aria-label="Step 2"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      2
    </span>
    <span
      aria-label="Step 3"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      3
    </span>
    <span
      aria-label="Step 4"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      4
    </span>
    <span
      aria-label="Step 5"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      5
    </span>
    <span
      aria-label="Step 6"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      6
    </span>
    <span
      aria-label="Step 7"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      7
    </span>
    <span
      aria-label="Step 8"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      8
    </span>
    <span
      aria-label="Step 9"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      9
    </span>
    <span
      aria-label="Step 10"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      10
    </span>
    <span
      aria-label="Step 21"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      21
    </span>
    <span
      aria-label="Step 32"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      32
    </span>
    <span
      aria-label="Step 43"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      43
    </span>
    <span
      aria-label="Step 54"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      54
    </span>
    <span
      aria-label="Step 65"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      65
    </span>
    <span
      aria-label="Step 76"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      76
    </span>
    <span
      aria-label="Step 87"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      87
    </span>
    <span
      aria-label="Step 98"
      class="text text--body-sm number-icon number-icon--base number-icon--lg"
      role="img"
    >
      98
    </span>
  </div>
</div>
`;

exports[`<NumberIcon /> Incomplete story renders snapshot 1`] = `
<div
  class="p-8"
>
  <span
    aria-label="Step 1"
    class="text text--body-xs number-icon number-icon--base number-icon--sm number-icon--incomplete"
    role="img"
  >
    <svg
      class="icon number-icon__icon"
      fill="inherit"
      height="0.25rem"
      role="img"
      style="--icon-size: 0.25rem;"
      viewBox="0 0 24 24"
      width="0.25rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Incomplete
      </title>
      <path
        d="M12 22C10.6167 22 9.31667 21.7373 8.1 21.212C6.88333 20.6873 5.825 19.975 4.925 19.075C4.025 18.175 3.31267 17.1167 2.788 15.9C2.26267 14.6833 2 13.3833 2 12C2 10.6167 2.26267 9.31667 2.788 8.1C3.31267 6.88333 4.025 5.825 4.925 4.925C5.825 4.025 6.88333 3.31233 8.1 2.787C9.31667 2.26233 10.6167 2 12 2C13.3833 2 14.6833 2.26233 15.9 2.787C17.1167 3.31233 18.175 4.025 19.075 4.925C19.975 5.825 20.6873 6.88333 21.212 8.1C21.7373 9.31667 22 10.6167 22 12C22 13.3833 21.7373 14.6833 21.212 15.9C20.6873 17.1167 19.975 18.175 19.075 19.075C18.175 19.975 17.1167 20.6873 15.9 21.212C14.6833 21.7373 13.3833 22 12 22Z"
      />
    </svg>
  </span>
</div>
`;

exports[`<NumberIcon /> NumberIconList story renders snapshot 1`] = `
<div
  class="p-8"
>
  <div
    class="flex flex-wrap gap-1"
  >
    <span
      aria-label="Item 1"
      class="text text--body-xs number-icon number-icon--base number-icon--sm"
      role="img"
    >
      1
    </span>
    <span
      aria-label="Item 2"
      class="text text--body-xs number-icon number-icon--base number-icon--sm number-icon--incomplete"
      role="img"
    >
      <svg
        class="icon number-icon__icon"
        fill="inherit"
        height="0.25rem"
        role="img"
        style="--icon-size: 0.25rem;"
        viewBox="0 0 24 24"
        width="0.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          incomplete step 2
        </title>
        <path
          d="M12 22C10.6167 22 9.31667 21.7373 8.1 21.212C6.88333 20.6873 5.825 19.975 4.925 19.075C4.025 18.175 3.31267 17.1167 2.788 15.9C2.26267 14.6833 2 13.3833 2 12C2 10.6167 2.26267 9.31667 2.788 8.1C3.31267 6.88333 4.025 5.825 4.925 4.925C5.825 4.025 6.88333 3.31233 8.1 2.787C9.31667 2.26233 10.6167 2 12 2C13.3833 2 14.6833 2.26233 15.9 2.787C17.1167 3.31233 18.175 4.025 19.075 4.925C19.975 5.825 20.6873 6.88333 21.212 8.1C21.7373 9.31667 22 10.6167 22 12C22 13.3833 21.7373 14.6833 21.212 15.9C20.6873 17.1167 19.975 18.175 19.075 19.075C18.175 19.975 17.1167 20.6873 15.9 21.212C14.6833 21.7373 13.3833 22 12 22Z"
        />
      </svg>
    </span>
    <span
      aria-label="Item 3"
      class="text text--body-xs number-icon number-icon--base number-icon--sm"
      role="img"
    >
      3
    </span>
    <span
      aria-label="Item 4"
      class="text text--body-xs number-icon number-icon--base number-icon--sm number-icon--incomplete"
      role="img"
    >
      <svg
        class="icon number-icon__icon"
        fill="inherit"
        height="0.25rem"
        role="img"
        style="--icon-size: 0.25rem;"
        viewBox="0 0 24 24"
        width="0.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          incomplete step 4
        </title>
        <path
          d="M12 22C10.6167 22 9.31667 21.7373 8.1 21.212C6.88333 20.6873 5.825 19.975 4.925 19.075C4.025 18.175 3.31267 17.1167 2.788 15.9C2.26267 14.6833 2 13.3833 2 12C2 10.6167 2.26267 9.31667 2.788 8.1C3.31267 6.88333 4.025 5.825 4.925 4.925C5.825 4.025 6.88333 3.31233 8.1 2.787C9.31667 2.26233 10.6167 2 12 2C13.3833 2 14.6833 2.26233 15.9 2.787C17.1167 3.31233 18.175 4.025 19.075 4.925C19.975 5.825 20.6873 6.88333 21.212 8.1C21.7373 9.31667 22 10.6167 22 12C22 13.3833 21.7373 14.6833 21.212 15.9C20.6873 17.1167 19.975 18.175 19.075 19.075C18.175 19.975 17.1167 20.6873 15.9 21.212C14.6833 21.7373 13.3833 22 12 22Z"
        />
      </svg>
    </span>
    <span
      aria-label="Item 5"
      class="text text--body-xs number-icon number-icon--base number-icon--sm number-icon--incomplete"
      role="img"
    >
      <svg
        class="icon number-icon__icon"
        fill="inherit"
        height="0.25rem"
        role="img"
        style="--icon-size: 0.25rem;"
        viewBox="0 0 24 24"
        width="0.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          incomplete step 5
        </title>
        <path
          d="M12 22C10.6167 22 9.31667 21.7373 8.1 21.212C6.88333 20.6873 5.825 19.975 4.925 19.075C4.025 18.175 3.31267 17.1167 2.788 15.9C2.26267 14.6833 2 13.3833 2 12C2 10.6167 2.26267 9.31667 2.788 8.1C3.31267 6.88333 4.025 5.825 4.925 4.925C5.825 4.025 6.88333 3.31233 8.1 2.787C9.31667 2.26233 10.6167 2 12 2C13.3833 2 14.6833 2.26233 15.9 2.787C17.1167 3.31233 18.175 4.025 19.075 4.925C19.975 5.825 20.6873 6.88333 21.212 8.1C21.7373 9.31667 22 10.6167 22 12C22 13.3833 21.7373 14.6833 21.212 15.9C20.6873 17.1167 19.975 18.175 19.075 19.075C18.175 19.975 17.1167 20.6873 15.9 21.212C14.6833 21.7373 13.3833 22 12 22Z"
        />
      </svg>
    </span>
    <span
      aria-label="Item 6"
      class="text text--body-xs number-icon number-icon--base number-icon--sm number-icon--incomplete"
      role="img"
    >
      <svg
        class="icon number-icon__icon"
        fill="inherit"
        height="0.25rem"
        role="img"
        style="--icon-size: 0.25rem;"
        viewBox="0 0 24 24"
        width="0.25rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          incomplete step 6
        </title>
        <path
          d="M12 22C10.6167 22 9.31667 21.7373 8.1 21.212C6.88333 20.6873 5.825 19.975 4.925 19.075C4.025 18.175 3.31267 17.1167 2.788 15.9C2.26267 14.6833 2 13.3833 2 12C2 10.6167 2.26267 9.31667 2.788 8.1C3.31267 6.88333 4.025 5.825 4.925 4.925C5.825 4.025 6.88333 3.31233 8.1 2.787C9.31667 2.26233 10.6167 2 12 2C13.3833 2 14.6833 2.26233 15.9 2.787C17.1167 3.31233 18.175 4.025 19.075 4.925C19.975 5.825 20.6873 6.88333 21.212 8.1C21.7373 9.31667 22 10.6167 22 12C22 13.3833 21.7373 14.6833 21.212 15.9C20.6873 17.1167 19.975 18.175 19.075 19.075C18.175 19.975 17.1167 20.6873 15.9 21.212C14.6833 21.7373 13.3833 22 12 22Z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`<NumberIcon /> Small story renders snapshot 1`] = `
<div
  class="p-8"
>
  <span
    aria-label="Step 1"
    class="text text--body-xs number-icon number-icon--base number-icon--sm"
    role="img"
  >
    1
  </span>
</div>
`;

exports[`<NumberIcon /> Success story renders snapshot 1`] = `
<div
  class="p-8"
>
  <span
    aria-label="Step 1"
    class="text text--body-sm number-icon number-icon--success number-icon--lg"
    role="img"
  >
    1
  </span>
</div>
`;

exports[`<NumberIcon /> SuccessSmall story renders snapshot 1`] = `
<div
  class="p-8"
>
  <span
    aria-label="Step 1"
    class="text text--body-xs number-icon number-icon--success number-icon--sm"
    role="img"
  >
    1
  </span>
</div>
`;
